pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1045,6 +1048,10 @@ private CompletableFuture<Subscription> getDurableSubscription(String subscripti
         }
 
         Map<String, Long> properties = PersistentSubscription.getBaseCursorProperties(replicated);
+        if (readCompacted) {
+            properties = new HashMap<>(properties);
+            properties.put(READ_COMPACTED_CURSOR_PROPERTIES, 1L);
```

Comment:
Actually, it will persistent noting? If a consumer enabled readCompacted but haven't ack messages for a few minutes, we will also get a chance to skip the data if the broker rewind the cursor?